### PR TITLE
add Visible by User Group column to Users: Viewing Access Levels

### PR DESCRIPTION
**Pull request by pe7er**
This PR adds an extra column with information to **Users: Viewing Access Levels** to get a better overview of **what User Groups** have been assigned to **Having Viewing Access**.
# Testing Instructions

Use Joomla with Test English (GB) sample data.
## Before the PR

Go to Users > Access Levels
To view the **User Groups** that have access to a specific **Viewing Access Level**, 
you'll have to click each of the Viewing Access Levels.

viewing-access-levels-before

So to see all **User Groups** that have access to the **Viewing Access Level** **Special**, you've to click on "Special"
viewing-access-levels-before2
## After the PR

Go to Users > Access Levels
The list of **Viewing Access Levels** has an extra column with "Visible by User Group".

viewing-access-levels-after

### CODE REVIEW

**Comment by andrepereiradasilva** [comment]
nice one! will test when possible

**Comment by hans2103** [comment]
I have tested this item :white_check_mark: successfully on 56c63600b396a70edce29607c93221c6137f8b9e<br /><br />I can reproduce the issue before and after the patch.
I think it's a useful submission to the Joomla! core. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10224">issues.joomla.org/joomla-cms/10224</a>.</sub>

**Comment by pe7er** [comment]
Sorry, Travis was not happy & I had to correct some code style.

Thanks for testing @hans2103

**Comment by brianteeman** [comment]
I think it will be better to use the existing string "User Groups Having Viewing Access" for the column heading. We have enough confusion and confusing strings for the ACL as it is

**Comment by allrude** [comment]
Tested successfully, nice one Pe7er !!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10224">issues.joomla.org/joomla-cms/10224</a>.</sub>

**Comment by pe7er** [comment]
Good point @brianteeman, I've used an existing language string & removed the new one.

Thanks for testing @allrude !

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on db43a7e7e9853dfc0cf9b311c2fa093858930db1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10224">issues.joomla.org/joomla-cms/10224</a>.</sub>

**Comment by brianteeman** [comment]
Thanks @pe7er Travis still not happy though ;(

On 4 May 2016 at 12:19, Peter Martin <notifications@github.com> wrote:

> Good point @brianteeman <https://github.com/brianteeman>, I've used an
> existing language string & removed the new one.
>
> Thanks for testing @allrude <https://github.com/allrude> !
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10224#issuecomment-216832012>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by pe7er** [comment]
No, I know. 
I am not happy either as my attempts to install PHPCodeSniffer on my system have been unsuccessful so far.. :-)

File: administrator/components/com_users/helpers/users.php [hunk 1]
```diff
@@ -221,4 +221,37 @@ public static function getTwoFactorMethods()
 
 		return $options;
 	}
+
+	/**
+	 * Get a list of the User Groups for Viewing Access Levels
+	 *
+	 * @param   string  $rules  User Groups in JSON format
+	 *
+	 * @return  array  An array.
```
Comment (andrepereiradasilva) [inline administrator/components/com_users/helpers/users.php]: `@return  array  An array.`

i think you can do better here :wink:

File: administrator/components/com_users/views/levels/tmpl/default.php [hunk 2]
```diff
@@ -57,6 +57,9 @@
 						<th>
 							<?php echo JHtml::_('searchtools.sort', 'COM_USERS_HEADING_LEVEL_NAME', 'a.title', $listDirn, $listOrder); ?>
 						</th>
+						<th>
```
Comment (andrepereiradasilva) [inline administrator/components/com_users/views/levels/tmpl/default.php]: shouldn't be hidden in tablet and mobile view? `hidden-tablet hidden-phone` css classes

File: administrator/components/com_users/views/levels/tmpl/default.php [hunk 3]
```diff
@@ -109,6 +112,9 @@
 							<?php endif; ?>
 						</td>
 						<td>
```
Comment (andrepereiradasilva) [inline administrator/components/com_users/views/levels/tmpl/default.php]: same as above

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 4c770af79180e206f63697e71a0960615c6570e7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10224">issues.joomla.org/joomla-cms/10224</a>.</sub>

**Comment by brianteeman** [comment]
@pe7er a recent PR has introduced a merge conflict ;(<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10224">issues.joomla.org/joomla-cms/10224</a>.</sub>

**Comment by pe7er** [comment]
I've corrected the merge conflict with PR #10273.
Please close this PR. Thank you all for testing & suggestions!

**Comment by brianteeman** [comment]
closed

**Pull request closed without merging**